src/cpp/src/icontinuous_batching.cpp

```diff
@@ -50,7 +50,17 @@ ContinuousBatchingPipeline::IContinuousBatchingPipeline::generate(
         input_ids.reserve(prompts.size());
         timer.start();
         for (const std::string& prompt : prompts) {
-            input_ids.push_back(m_tokenizer.encode(prompt).input_ids);
+            ov::Tensor encoded_inputs;
+            try {
+                ChatHistory history({{{"role", "user"}, {"content", prompt}}});
+                constexpr bool add_generation_prompt = true;
+                auto templated_prompt = m_tokenizer.apply_chat_template(history, add_generation_prompt);
+                encoded_inputs = m_tokenizer.encode(templated_prompt, ov::genai::add_special_tokens(false)).input_ids;
+            } catch (const std::exception& error) {
```

Comment:
That would ignore errors caused by chat template. Like in the ticket 159466. I think you should only skip template if it's empty.

Reply:
added check if template is empty